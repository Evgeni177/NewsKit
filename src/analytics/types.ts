export type TealiumLinkEvent = {
  event_navigation_action: React.ReactNode;
  event_navigation_browsing_method?: string;
};

export type TealiumViewEvent = {
  page_url: string;
};

export type InteractionEvent = {
  type: 'link';
  data: TealiumLinkEvent;
};

export type PageViewEvent = {
  type: 'view';
  data: TealiumViewEvent;
};

export type AnalyticsEvent = InteractionEvent | PageViewEvent;

export type UTag = {
  link(e: TealiumLinkEvent): void;
  view(e: TealiumViewEvent): void;
};

export interface ExtendedWindow extends Window {
  utag: UTag;
}
