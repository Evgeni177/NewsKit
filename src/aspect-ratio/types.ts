import {
  MaxHeightProperty,
  MaxWidthProperty,
  PaddingTopProperty,
  HeightProperty,
  WidthProperty,
} from 'csstype';
import {GetAspectRatioProp} from '../utils/get-aspect-ratio';

export interface ContainerProp {
  paddingTop?: PaddingTopProperty<string>;
  maxHeight?: MaxHeightProperty<string>;
  maxWidth?: MaxWidthProperty<string>;
}

export interface StyledContainerProps {
  $height?: HeightProperty<string>;
  $width?: WidthProperty<string>;
}

export interface AspectRatioProps
  extends GetAspectRatioProp,
    Omit<ContainerProp, 'paddingTop'> {}
