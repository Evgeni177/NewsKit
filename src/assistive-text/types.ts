import {EnhancerOverrides} from '../form/types';
import {MQ} from '../utils/style';
import {WithEnhancersProps} from '../with-enhancers/types';

export type AssistiveTextProps = React.HTMLAttributes<HTMLParagraphElement> &
  Omit<
    WithEnhancersProps,
    'enhancersType' | 'isFocused' | 'componentDefaultsPath'
  > & {
    overrides?: EnhancerOverrides & {
      stylePreset?: MQ<string>;
      typographyPreset?: MQ<string>;
      minHeight?: MQ<string>;
    };
  };
