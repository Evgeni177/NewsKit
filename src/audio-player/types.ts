import {TrackControlProps, ControlsOverrideProps} from './controls';
import {SliderOverrideProps} from '../slider';
import {VolumeControlOverrideProps} from '../volume-control';
import {StylePresetKeys} from '../theme';

export interface AudioPlayerProps
  extends Omit<React.AudioHTMLAttributes<HTMLAudioElement>, 'controls'>,
    TrackControlProps {
  captionSrc?: string;
  popoutHref?: string;
  live?: boolean;
  ariaLandmark?: string;
  overrides?: {
    seekBar?: {
      slider?: SliderOverrideProps;
      buffering?: {
        stylePreset?: StylePresetKeys;
      };
    };
    controls?: ControlsOverrideProps;
    volumeControl?: VolumeControlOverrideProps;
  };
}
