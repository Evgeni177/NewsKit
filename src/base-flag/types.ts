import {SizingKeys, IconSizeKeys} from '../themes';
import {PaddingPresetKeys} from '../themes/mappers/spacing';

export interface BaseFlagProps {
  $typePreset?: string;
  $stylePreset?: string;
  isDisabled?: boolean;
  borderRadiusSize?: SizingKeys;
  minHeight?: SizingKeys;
  padding?: PaddingPresetKeys;
  iconSize?: IconSizeKeys;
}
