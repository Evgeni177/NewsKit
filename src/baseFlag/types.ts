import {TagSize} from '../tag/utils';
import {FlagSize} from '../flag/utils';

export interface BaseFlagProps {
  disabled?: boolean;
  $size?: FlagSize | TagSize;
  $stylePreset?: string;
}
