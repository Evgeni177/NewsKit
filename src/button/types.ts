import {ColorKeys} from '../themes';
import {SizingKeys} from '../themes/newskit-light/spacing';
import {ThemeProp} from '../utils/style';

export type StylePresetAndTheme = Pick<ButtonProps, '$stylePreset'> & ThemeProp;
export interface ButtonProps {
  $size?: ButtonSize;
  $iconColor?: ColorKeys;
  $stylePreset?: string;
}

export interface ButtonSizing {
  paddingX: SizingKeys;
  paddingY: SizingKeys;
  width?: SizingKeys;
  height?: SizingKeys;
}

export enum ButtonSize {
  Small = 'small',
  Medium = 'medium',
  Large = 'large',
}
