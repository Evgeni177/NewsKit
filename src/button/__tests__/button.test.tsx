import {renderToFragmentWithTheme} from '../../test/test-utils';
import {Button} from '..';
import {ButtonSize, IconPlacement} from '../types';
import {Email} from '../../icons';
import {ColorKeys} from '../../themes/mappers/colors';

describe('Button', () => {
  test('renders with fallback formatting', () => {
    const fragment = renderToFragmentWithTheme(Button);
    expect(fragment).toMatchSnapshot();
  });

  test('renders Small Button', () => {
    const props = {
      $size: ButtonSize.Small,
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders Medium Button', () => {
    const props = {
      $size: ButtonSize.Medium,
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders Large Button', () => {
    const props = {
      $size: ButtonSize.Large,
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders with different style preset', () => {
    const props = {
      $stylePreset: 'interactive050',
    };

    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders Icon Button', () => {
    const props = {
      $size: ButtonSize.Large,
      icon: Email,
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders when there is no icon', () => {
    const props = {
      $size: ButtonSize.Small,
      children: 'click this!',
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders when there is an icon with a colour', () => {
    const props = {
      $size: ButtonSize.Small,
      icon: Email,
      $iconColor: 'buttonText' as ColorKeys,
      children: 'click this!',
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders when there is an icon with the placement set to start', () => {
    const props = {
      $size: ButtonSize.Small,
      icon: Email,
      iconPlacement: IconPlacement.Start,
      children: 'click this!',
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });

  test('renders when there is an icon with the placement set to end', () => {
    const props = {
      $size: ButtonSize.Small,
      icon: Email,
      iconPlacement: IconPlacement.End,
      children: 'click this!',
    };
    const fragment = renderToFragmentWithTheme(Button, props);
    expect(fragment).toMatchSnapshot();
  });
});
