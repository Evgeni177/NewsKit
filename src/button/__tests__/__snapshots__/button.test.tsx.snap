// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Icon Button 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-3 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 16px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 16px;
  line-height: 1.75;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 16px 24px;
}

.emotion-3 svg {
  fill: #FFFFFF;
  color: #FFFFFF;
}

.emotion-3:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-3:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-3:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-3:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-3:disabled svg {
  fill: #c0c0c0;
  color: #c0c0c0;
}

.emotion-3:disabled {
  cursor: not-allowed;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.emotion-0 {
  display: inline-block;
  fill: #0A68C1;
  color: #0A68C1;
  width: 24px;
  height: 24px;
}

<button
    class="emotion-3"
    type="button"
  >
    <div
      class="emotion-2"
      wrap="nowrap"
    >
      <div
        class="emotion-1"
        wrap="nowrap"
      >
        <svg
          class="emotion-0"
          viewBox="0 0 24 24"
        >
          <title>
            Email icon
          </title>
          <path
            d="M20,4 L4,4 C2.9,4 2.01,4.9 2.01,6 L2,18 C2,19.1 2.9,20 4,20 L20,20 C21.1,20 22,19.1 22,18 L22,6 C22,4.9 21.1,4 20,4 L20,4 Z M20,8 L12,13 L4,8 L4,6 L12,11 L20,6 L20,8 L20,8 Z"
          />
        </svg>
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button renders Large Button 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 16px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 16px;
  line-height: 1.75;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 16px 24px;
}

.emotion-1:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-1:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-1:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-1:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

<button
    class="emotion-1"
    type="button"
  >
    <div
      class="emotion-0"
      wrap="nowrap"
    />
  </button>
</DocumentFragment>
`;

exports[`Button renders Medium Button 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 12px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 14px;
  line-height: 1.7142857142857142;
  font-weight: 500;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  padding: 12px 16px;
}

.emotion-1:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-1:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-1:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-1:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

<button
    class="emotion-1"
    type="button"
  >
    <div
      class="emotion-0"
      wrap="nowrap"
    />
  </button>
</DocumentFragment>
`;

exports[`Button renders Small Button 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
}

.emotion-1:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-1:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-1:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-1:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<button
    class="emotion-1"
    type="button"
  >
    <div
      class="emotion-0"
      wrap="nowrap"
    />
  </button>
</DocumentFragment>
`;

exports[`Button renders when there is an icon with a colour 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
}

.emotion-4 svg {
  fill: #FFFFFF;
  color: #FFFFFF;
}

.emotion-4:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-4:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-4:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-4:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-4:disabled svg {
  fill: #c0c0c0;
  color: #c0c0c0;
}

.emotion-4:disabled {
  cursor: not-allowed;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-0 {
  display: inline-block;
  fill: #FFFFFF;
  color: #FFFFFF;
  width: 16px;
  height: 16px;
}

<button
    class="emotion-4"
    type="button"
  >
    <div
      class="emotion-3"
      wrap="nowrap"
    >
      <div
        class="emotion-1"
        wrap="nowrap"
      >
        <svg
          class="emotion-0"
          viewBox="0 0 24 24"
        >
          <title>
            Email icon
          </title>
          <path
            d="M20,4 L4,4 C2.9,4 2.01,4.9 2.01,6 L2,18 C2,19.1 2.9,20 4,20 L20,20 C21.1,20 22,19.1 22,18 L22,6 C22,4.9 21.1,4 20,4 L20,4 Z M20,8 L12,13 L4,8 L4,6 L12,11 L20,6 L20,8 L20,8 Z"
          />
        </svg>
      </div>
      <div
        class="emotion-1"
        wrap="nowrap"
      >
        click this!
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button renders when there is an icon with the placement set to end 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
}

.emotion-4 svg {
  fill: #FFFFFF;
  color: #FFFFFF;
}

.emotion-4:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-4:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-4:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-4:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-4:disabled svg {
  fill: #c0c0c0;
  color: #c0c0c0;
}

.emotion-4:disabled {
  cursor: not-allowed;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-1 {
  display: inline-block;
  fill: #0A68C1;
  color: #0A68C1;
  width: 16px;
  height: 16px;
}

<button
    class="emotion-4"
    type="button"
  >
    <div
      class="emotion-3"
      wrap="nowrap"
    >
      <div
        class="emotion-0"
        wrap="nowrap"
      >
        click this!
      </div>
      <div
        class="emotion-0"
        wrap="nowrap"
      >
        <svg
          class="emotion-1"
          viewBox="0 0 24 24"
        >
          <title>
            Email icon
          </title>
          <path
            d="M20,4 L4,4 C2.9,4 2.01,4.9 2.01,6 L2,18 C2,19.1 2.9,20 4,20 L20,20 C21.1,20 22,19.1 22,18 L22,6 C22,4.9 21.1,4 20,4 L20,4 Z M20,8 L12,13 L4,8 L4,6 L12,11 L20,6 L20,8 L20,8 Z"
          />
        </svg>
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button renders when there is an icon with the placement set to start 1`] = `
<DocumentFragment>
  .emotion-4 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
}

.emotion-4 svg {
  fill: #FFFFFF;
  color: #FFFFFF;
}

.emotion-4:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-4:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-4:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-4:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-4:disabled svg {
  fill: #c0c0c0;
  color: #c0c0c0;
}

.emotion-4:disabled {
  cursor: not-allowed;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: calc(-8px/2);
  margin-right: calc(-8px/2);
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-left: calc(8px/2);
  margin-right: calc(8px/2);
}

.emotion-0 {
  display: inline-block;
  fill: #0A68C1;
  color: #0A68C1;
  width: 16px;
  height: 16px;
}

<button
    class="emotion-4"
    type="button"
  >
    <div
      class="emotion-3"
      wrap="nowrap"
    >
      <div
        class="emotion-1"
        wrap="nowrap"
      >
        <svg
          class="emotion-0"
          viewBox="0 0 24 24"
        >
          <title>
            Email icon
          </title>
          <path
            d="M20,4 L4,4 C2.9,4 2.01,4.9 2.01,6 L2,18 C2,19.1 2.9,20 4,20 L20,20 C21.1,20 22,19.1 22,18 L22,6 C22,4.9 21.1,4 20,4 L20,4 Z M20,8 L12,13 L4,8 L4,6 L12,11 L20,6 L20,8 L20,8 Z"
          />
        </svg>
      </div>
      <div
        class="emotion-1"
        wrap="nowrap"
      >
        click this!
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button renders when there is no icon 1`] = `
<DocumentFragment>
  .emotion-2 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
}

.emotion-2:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-2:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-2:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-2:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-2:disabled {
  cursor: not-allowed;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<button
    class="emotion-2"
    type="button"
  >
    <div
      class="emotion-1"
      wrap="nowrap"
    >
      <div
        class="emotion-0"
        wrap="nowrap"
      >
        click this!
      </div>
    </div>
  </button>
</DocumentFragment>
`;

exports[`Button renders with different style preset 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  border-style: solid;
  border-color: #C8E4FD;
  border-width: 1px;
  color: #0A68C1;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
}

.emotion-1:hover:not(:disabled) {
  background-color: #F6FBFF;
}

.emotion-1:active:not(:disabled) {
  background-color: #F6FBFF;
  border-width: #0A68C1;
}

.emotion-1:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

<button
    class="emotion-1"
    type="button"
  >
    <div
      class="emotion-0"
      wrap="nowrap"
    />
  </button>
</DocumentFragment>
`;

exports[`Button renders with fallback formatting 1`] = `
<DocumentFragment>
  .emotion-1 {
  display: inline-block;
  position: relative;
  width: auto;
  border: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  overflow: hidden;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  -webkit-transition-duration: 200ms;
  transition-duration: 200ms;
  -webkit-transition-timing-function: cubic-bezier(0,0,.5,1);
  transition-timing-function: cubic-bezier(0,0,.5,1);
  cursor: pointer;
  background-color: #0A68C1;
  color: #FFFFFF;
  border-radius: 8px;
  min-height: 8px;
  font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
  font-size: 12px;
  line-height: 1.6666666666666667;
  font-weight: 500;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  padding: 8px 12px;
}

.emotion-1:hover:not(:disabled) {
  background-color: #02509B;
}

.emotion-1:active:not(:disabled) {
  background-color: #01396E;
}

.emotion-1:focus:not(:disabled) {
  border-width: 2px;
}

.emotion-1:disabled {
  background-color: #f0f0f0;
  color: #c0c0c0;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<button
    class="emotion-1"
    type="button"
  >
    <div
      class="emotion-0"
      wrap="nowrap"
    />
  </button>
</DocumentFragment>
`;
